### PR TITLE
Forgot password

### PR DESCRIPTION
**Pull request by BMartinos**
This PR relates to issue: #310

### CODE REVIEW

**Comment by BMartinos** [comment]
Hey @hnnesv Would you mind reviewing this PR for me when you get a chance :). Thanks. This PR is dependant on https://github.com/jembi/openhim-core-js/pull/624

**Comment by BMartinos** [comment]
@hnnesv these PR's are now good to go :)

**Comment by hnnesv** [comment]
Thanks @BMartinos I'm proceeding with this now. Just a few first thoughts [both are nice-to-haves]:
* after setting your new password - could you redirect the user directly to the login page with a message saying success on there, instead of the user needing to click 'Proceed to login'? Or have a timer after which the user is redirected?
* would it be possible that when you click on 'forgot your password' and you've filled in an email address on the login page, that address automatically gets populated in the email request box on the forgot page?

But otherwise functionally it works really well!

File: app/scripts/controllers/setPassword.js [hunk 1]
```diff
@@ -16,13 +16,14 @@ angular.module('openhimConsoleApp')
...
       $scope.user = user;
+      $scope.tokenType = user.tokenType;
     };
 
     var setPassError = function(err){
       Alerting.AlertReset();
       // on error - add server error alert
       if ( err.status === 404 ){
-        Alerting.AlertAddMsg('top', 'danger', 'No user exist for the supplied token. Please ensure you have copied the token correctly.');
+        Alerting.AlertAddMsg('top', 'danger', 'The supplied token does not exit.');
```
Comment (hnnesv) [inline app/scripts/controllers/setPassword.js]: how about simply just "Invalid token"

**Comment by hnnesv** [comment]
Looks good to me - just some comments: if you've got space we can look at the above suggestions, but it's not too important. Otherwise :+1:

**Comment by BMartinos** [comment]
Hey @hnnesv I have made all the updates as mentioned :). This is ready for review again. Thanks

**Comment by hnnesv** [comment]
:+1: Thanks @BMartinos looks good!

**Comment by BMartinos** [comment]
Hey @hnnesv ive made a few small changes to this PR if you can maybe just have a quick look at it again. Thanks

**Comment by hnnesv** [comment]
@BMartinos thanks, looks good! :+1:

**Pull request merged**